src/test/java/testapp/adapter/http/UserServiceTest.java

```diff
@@ -0,0 +1,136 @@
+package testapp.adapter.http;
+
+
+import com.google.sitebricks.client.transport.Json;
+import com.google.sitebricks.headless.Reply;
+import com.google.sitebricks.headless.Request;
+import org.jmock.Expectations;
+import org.jmock.integration.junit4.JUnitRuleMockery;
+import org.junit.Test;
+import testapp.adapter.memory.InMemoryUserRepository;
+import testapp.core.User;
+import testapp.core.UserDto;
+import testapp.core.UserRepository;
+
+import java.util.Collections;
+import java.util.LinkedList;
+
+import static org.junit.Assert.*;
+import static testapp.adapter.http.JsonBuilder.aNewJson;
+import static testapp.adapter.http.JsonBuilder.aNewJsonArray;
+import static testapp.adapter.http.Matchers.containsJson;
+import static testapp.adapter.http.Matchers.isBadRequest;
+import static testapp.adapter.http.Matchers.isOk;
+import static testapp.adapter.http.Matchers.returns;
+
+/**
+ * @author Vasil Mitov <v.mitov.clouway@gmail.com>
+ */
+public class UserServiceTest {
+  public JUnitRuleMockery context = new JUnitRuleMockery();
+  public final Request request = context.mock(Request.class);
+
+  @Test
+  public void registerUser() throws Exception {
+    UserRepository userRepository = new InMemoryUserRepository();
+    UserService userService = new UserService(userRepository);
+    final UserDto userDto = new UserDto(1, "Vasil", Collections.<Integer>emptyList());
+
+    context.checking(new Expectations() {{
+      oneOf(request).read(UserDto.class);
+      will(returnValue(new RequestReadOf(userDto)));
+    }});
+    Reply<?> registerReply = userService.registerUser(request);
+
+    assertThat(registerReply, isOk());
+  }
+
+  @Test
+  public void userAlreadyRegistered() throws Exception {
+    UserRepository userRepository = new InMemoryUserRepository();
+    UserService userService = new UserService(userRepository);
+    userRepository.register(new User(1, "Vasil", Collections.<Integer>emptyList()));
+    final UserDto userDto = new UserDto(1, "Vasil", Collections.<Integer>emptyList());
+
+    context.checking(new Expectations() {{
+      oneOf(request).read(UserDto.class);
+      will(returnValue(new RequestReadOf(userDto)));
+    }});
+    Reply<?> registerReply = userService.registerUser(request);
```

Comment:
well, you wont have to mock it here and will not have this expectations above for one